File: Two Number Sum/TwoNumberSum.java
import java.util.*;

class TwoNumberSum {
    public static int[] twoNumberSum(int[] array, int targetSum) {
        if (array.length < 2) {
            return new int[0];
        }
        int diff;
        Map<Integer, Integer> hashTable = new HashMap<>();
        for (int element : array) {
            diff = targetSum - element;
            if (hashTable.containsKey(diff)) {
                return new int[] { diff, element };
            }
            hashTable.put(element, diff);
        }
        return new int[0];
    }
}

File: Two Number Sum/TwoNumberSum.kt
package com.algoexpert.program

fun twoNumberSum(array: MutableList<Int>, targetSum: Int): List<Int> {
    if (array.size < 2) {
        return listOf<Int>()
    }
    var diff: Int
    val map = mutableMapOf<Int, Boolean>()
    for (num in array) {
        diff = targetSum - num
        if (map.containsKey(diff)) {
            return listOf<Int>(diff, num)
        }
        map[num] = true
    }
    return listOf<Int>()
}
